contracts/interchain-token-service/src/contract/execute/interceptors.rs

```diff
@@ -299,10 +299,17 @@ pub fn register_custom_token(
         register_token.token_address.clone(),
     )
     .change_context(Error::State)?;
-    ensure!(
-        existing_token.is_none(),
-        Error::TokenAlreadyRegistered(register_token.token_address)
-    );
+
+    if let Some(registered_token) = existing_token {
+        ensure!(
+            registered_token.decimals == register_token.decimals,
```

Comment:
```suggestion
    if let Some(existing_token) = existing_token {
        ensure!(
            existing_token == register_token,
```
check the full struct for in case other fields are added later?